frontend/src/features/admin-form/settings/SettingsPage.tsx

```diff
@@ -60,55 +63,56 @@ export const SettingsPage = (): JSX.Element => {
       navigate(`${ADMINFORM_ROUTE}/${formId}/${ADMINFORM_RESULTS_SUBROUTE}`)
   }, [formId, hasEditAccess, isCollabLoading, navigate])
 
-  const tabConfig = useMemo(() => {
-    const emailsNotificationsTab =
-      settings?.responseMode === FormResponseMode.Encrypt ||
-      settings?.responseMode === FormResponseMode.Email
-        ? {
-            label: 'Email notifications',
-            icon: BiMailSend,
-            component: SettingsEmailsPage,
-            path: 'email-notifications',
-            showRedDot: true,
-          }
-        : null
-
-    const baseConfig: (TabEntry | null)[] = [
-      {
-        label: 'General',
-        icon: BiCog,
-        component: SettingsGeneralPage,
-        path: 'general',
-      },
-      {
-        label: 'Singpass',
-        icon: BiKey,
-        component: SettingsAuthPage,
-        path: 'singpass',
-      },
-      emailsNotificationsTab,
-      {
-        label: 'Twilio credentials',
-        icon: BiMessage,
-        component: SettingsTwilioPage,
-        path: 'twilio-credentials',
-      },
-      {
-        label: 'Webhooks',
-        icon: BiCodeBlock,
-        component: SettingsWebhooksPage,
-        path: 'webhooks',
-      },
-      {
-        label: 'Payments',
-        icon: BiDollar,
-        component: SettingsPaymentsPage,
-        path: 'payments',
-      },
-    ]
-
-    return baseConfig.filter(isNonEmpty)
-  }, [settings?.responseMode])
+  const isTest = process.env.NODE_ENV === 'test'
+  // For beta flagging email notifications tab.
+  const emailNotificationsTab =
+    isUserLoading ||
+    isFormSettingLoading ||
```

Comment:
Want to understand if we actually need to explicitly check on `isUserLoading || isFormSettingLoading`

Wouldn't `!user?`  manage the situation where the data isn't ready yet i.e., `isUserLoading`? (similarly for `!formSettings?`)

Reply:
```
 const emailNotificationsTab =
    isUserLoading ||
    isFormSettingLoading ||
    // TODO: (MRF-email-notif) Remove isTest and betaFlag check when MRF email notifications is out of beta
    (!isTest &&
      formSettings?.responseMode === FormResponseMode.Multirespondent &&
      !user?.betaFlags?.mrfEmailNotifications)
      ? null
      : {
          label: 'Email notifications',
          icon: BiMailSend,
          component: SettingsEmailsPage,
          path: 'email-notifications',
          showRedDot: true,
        }
```

For this part of the code, when the `formSettings` is not loaded yet, `formSettings?.responseMode === FormResponseMode.Multirespondent ` evaluates to `falsy` value which causes the tab to be included. 

When the `formSettings` are finally loaded, then it will see it is not MRF and then the tab disappears. 

This causes a flickering issue. 

Alternatively, I could use `!formSettings` -> `null` as the check to replace `isFormSettingLoading` if desired. However, I felt that using the `isLoading` checks more accurately explains what this clause is doing and might be more readable and hence I chose that.

Reply:
For example, I could change it to: 
```
 const emailNotificationsTab =
    !user ||
    !formSettings ||
    // TODO: (MRF-email-notif) Remove isTest and betaFlag check when MRF email notifications is out of beta
    (!isTest &&
      formSettings.responseMode === FormResponseMode.Multirespondent &&
      !user.betaFlags?.mrfEmailNotifications)
      ? null
      : {
          label: 'Email notifications',
          icon: BiMailSend,
          component: SettingsEmailsPage,
          path: 'email-notifications',
          showRedDot: true,
        }
```